packages/react-adminer/src/components/AddToFavoriteButton.tsx

```diff
@@ -0,0 +1,111 @@
+import { Button, Input, notification } from 'antd';
+import { useState } from 'react';
+import Modal from './Modal';
+
+interface Props {
+	where: Record<string, any> | undefined;
+	filterConfig: any;
+}
+
+const AddToFavoriteButton: React.FC<Props> = ({ where, filterConfig }) => {
+	const [isModalVisible, setIsModalVisible] = useState(false);
+	const [favoriteFilterName, setFavoriteFilterName] = useState('');
+
+	const showModal = (): any => {
+		setIsModalVisible(true);
+	};
+
+	const checkDuplicityFavouriteFilter = (
+		actualLocalStorage: any,
+		name: string,
+		filter: Record<string, any> | undefined,
+	): void => {
+		const storageObj = JSON.parse(actualLocalStorage);
+		let isDuplicated = false;
+		storageObj.favourites.forEach((e: any, index: number) => {
+			if (!isDuplicated) {
+				if (e.name) {
```

Comment:
Dvě podmínky za sebou znamenají většinou že to lze napsat do jedný. U moderných jazyků nemusíš kontrolovat promennou (name). Tady ti staci `e?.name ===` a rikas tim vlastne "pokud existuje name, tak ho porovnej s ...".